### DISCUSSION TITLE
Evaluate a material property/aux variable at arbitrary location using interpolation

### DISCUSSION THREAD

**Discussion by chunhuizhao478** [idaholab/moose#26747] [Q&A General]
Hi all, I would like to do something similar as in `DiscreteNucleation`, but the nuclei location is independent of quadrature points, the nucleation is energy-based by evaluating the shear strain energy density involving stress tensor. I would like to get stress tensor values at those locations using interpolation. I wonder if there is an easy way to achieve it, thanks!

**Comment by GiudGiud** [comment]
Hello

The stress tensor is a tensor material property right?
Is that the only quantity you want to evaluate at an arbitrary point?

**Reply by chunhuizhao478** [reply]
yes I think so, the formula is as follows:

```math
\phi = \frac{1}{2G} \sigma_{ij}'(x^k) \sigma_{ij}'(x^k), \sigma_{ij}'= \sigma_{ij} - \frac{1}{3} \sigma_{kk} \delta_{ij}
```

$x^k$ is the location of the nuclei

**Reply by GiudGiud** [reply]
what kind of interpolation do you want?
Do you think you could compute it from the location of the quadrature points and the values at those quadrature points?

**Reply by chunhuizhao478** [reply]
So the energy of the nuclei is the criterion to nucleate the nuclei ($\phi > \phi_c = const$), I prefer not to restrict the placement of nuclei by the quadrature point locations. If I use higher-order elements for the field, given the qp values within the element, it should be able to compute the value at a point within this element by interpolation.

I have the implementation in FENICS before, which handles this quite easily: `_stress->eval(point_value,point_locs);`, I'm not sure if MOOSE has something similar.

**Reply by GiudGiud** [reply]
If you were using a functor material property instead of a regular material property, we have exactly these kind of evaluations at arbirtrary point implemented. 

>  If I use higher-order elements for the field

the field of stress tensors, which FE family and order is it supposed to be? If any

**Reply by chunhuizhao478** [reply]
Thanks @GiudGiud. I think the stress field can be `FIRST MONOMINAL `, not high-order for now.

**Reply by GiudGiud** [reply]
I would project it onto a few auxiliary variables of type FIRST MONOMIAL for each component then.

Once you have that done, you can evaluate them at an arbitrary location using the Functor APIs:
see the ElemPointArg here
https://mooseframework.inl.gov/syntax/Functors/index.html

**Reply by chunhuizhao478** [reply]
Thanks @GiudGiud, I would give it a try.
